dimod/sampleset.py

```diff
@@ -1391,6 +1391,81 @@ def slice(self, *slice_args, **kwargs):
         return type(self)(record, self.variables, copy.deepcopy(self.info),
                           self.vartype)
 
+    def append_data_vectors(self, vectors, inplace=True):
+        """Return the :class:`SampleSet` with additional fields in
+        :attr:`SampleSet.record`.
+
+        Args:
+            vectors (dict):
+                Each key is a new field name and each value is a list of 
+                per-sample data to be appended to :attr:`SampleSet.record`. 
+                Per-sample data should be scalar or numpy arrays (lists 
+                and tuples will be converted to numpy arrays).
+
+            inplace (bool, optional, default=True):
+                If True, the instantiated :class:`SampleSet` is updated; 
+                otherwise, a new :class:`SampleSet` is returned.
+
+        Returns:
+            :obj:`.SampleSet`: SampleSet. If `inplace` is True, returns itself.
+
+        Examples:
+            The following example appends a field of lists to :attr:`SampleSet.record`.
+
+            >>> import numpy as np
+            ...
+            >>> sampleset = dimod.SampleSet(np.rec.array([([-1,  1], -1.4, 1), ([-1,  1], -1.4, 1)],
+            ...                             dtype=[('sample', 'i1', (2,)), ('energy', '<f8'), 
+            ...                             ('num_occurrences', '<i8')]), [0, 1], {}, 'SPIN')
```

Comment:
Why not something lighter, like:
```suggestion
            >>> sampleset = dimod.SampleSet.from_samples([[-1,  1], [-1,  1]], energy=[1, 1], vartype='SPIN')
```
Then you also don't need numpy import.

But ignore if you were trying to visualize the record structure this way.